src/pages/home/sidebar/SidebarLinks.js

```diff
@@ -136,11 +91,19 @@ class SidebarLinks extends React.Component {
         SidebarUtils.setIsSidebarLoadedReady();
         this.isSidebarLoaded = true;
 
+        let modal = {};
+        this.unsubscribeOnyxModal = onyxSubscribe({
+            key: ONYXKEYS.MODAL,
+            callback: (modalArg) => {
+                modal = modalArg;
+            },
+        });
```

Comment:
This is one of the performance improvements. The modal state is needed for a logic, that has nothing to do with the render/UI output. Thus, the component shouldn't re-render (because it's a really expensive to render one) just because we open a modal.

Long term I want to remove the modal state from onyx and move it e.g. to a react context, as I think it doesn't belong in the onyx store.